runtime/jcl/common/shared.c

```diff
@@ -546,40 +543,68 @@ getPathProtocolFromURL(JNIEnv* env, jobject url, jmethodID URLgetPathID, jmethod
 	return rc;
 }
 
+static BOOLEAN
+isPathTypeJimage(const char *pathChars, jsize pathLen)
+{
+	char JimageEndsWith[] = DIR_SEPARATOR_STR "lib" DIR_SEPARATOR_STR "modules";
+	IDATA len = LITERAL_STRLEN(JimageEndsWith);
+
+	if (pathLen >= len) {
```

Comment:
The test should be `>`, not `>=` to not accept the absolute path `/lib/modules`.